### DISCUSSION TITLE
Molecules in supercells

### DISCUSSION THREAD

**Discussion by majordt** [materialsproject/pymatgen#2497] [Q&A] [answered]
I have a cif file of molecules, and I'm creating a supercell and I want to be able to access the coordinates of the molecules of the original cell (each molecule individually), so that I can calculate distance distributions to other molecules (e.g., within minimum image convention). How can I do this?
I tried this:

struct = Structure.from_file("227799.cif")
print(struct)
new_molecule = Molecule.from_sites(sorted(molecules[0], key=lambda site: site.species))
sg = StructureGraph.with_local_env_strategy(struct, JmolNN())
molecules = sg.get_subgraphs_as_molecules()

This only gives a single unique molecule (I want the central molecules in their original positions in the unit cell as molecular objects that I can manipulate).

Thanks,
Dan

**Comment by mkhorton** [comment, marked as answer]
Hi Dan,

This function is working as intended -- it is intended to extract a list of unique molecules.

The functionality you want does sound useful however. Perhaps you can take a look at the code in `get_subgraphs_as_molecules()` and see how it might be adapted for the functionality you want? I can imagine it already does most of what you need.

Hope this helps,

Matt

**Reply by majordt** [reply]
Hi Matt, I wrote a modified version of this function, which takes an additional argument ('unique'), and allows also returning all molecules in the crystal. It seems to work fine. 
Thanks for the suggestion,
Dan

**Reply by mkhorton** [reply]
Glad it worked out! Feel welcome to open a PR too if you want to contribute this code back if you think it'd be helpful to others
